screen/RoomMaker/RoomMakerScreen.js

```diff
@@ -1,13 +1,158 @@
-import React from 'react';
+import React, { useState, useReducer, useCallback } from 'react';
 import { View, Text, StyleSheet } from 'react-native';
+import DropDownPicker from 'react-native-dropdown-picker';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigation } from '@react-navigation/native';
 
+import { BASE_URL } from '@env';
+import RoomTitle from './components/RoomTitle';
+import GameLevel from './components/GameLevel';
 import COLORS from '../../common/constants/COLORS';
 import FONT from '../../common/constants/FONT';
+import showErrorMessage from '../../common/utils/showErrorMessage';
+import StandardModal from '../../common/components/StandardModal';
+import ActiveButton from '../../common/components/ActiveButton';
+import HttpClient from '../../service/http';
+import RoomService from '../../service/room';
+import { toggleModal } from '../../store/uiSlice';
+
+const httpClient = new HttpClient(BASE_URL);
+const roomService = new RoomService(httpClient);
+
+const formReducer = (state, action) => {
+  if (action.type === 'INPUT_CHANGE') {
+    const { gameMode } = action;
+    const updatedValue = action.input;
+    const updatedValidation = action.isValidValue;
+
+    return {
+      modeValue: gameMode,
+      inputValue: updatedValue,
+      isValid: updatedValidation,
+    };
+  }
+
+  return state;
+};
+
+const RoomMakerScreen = () => {
+  const navigation = useNavigation();
+  const dispatch = useDispatch();
+  const isModalVisible = useSelector((state) => state.ui.isModalVisible);
+
+  const [title, setTitle] = useState('');
+  const [open, setOpen] = useState(false);
+  const [mode, setMode] = useState('survival');
+  const [items, setItems] = useState([
+    { label: 'Survival', value: 'survival' },
+    { label: '1 : 1', value: 'oneOnOne' },
+  ]);
+
+  const handlePressedButton = () => {
```

Comment:
handle[Event] 에서 event의 동사 부분은 동사 원형으로 통일하는게 좋을 것 같습니다 
```suggestion
  const handlePresseButton = () => {
```